### ISSUE TITLE
[BUG] Error handling default field values in dataclasses as inputs

### ISSUE THREAD

**Issue by EngHabu** [flyteorg/flyte#5569]
### Describe the bug

Minimal repro:
```python
import typing
from dataclasses_json import dataclass_json
from flytekit import task, workflow

from dataclasses import dataclass


@dataclass_json
@dataclass
class User:
    name: str = None
    github_handle: str = None

@task
def single_item_task(user: User) -> User:
    return user


@workflow
def single_item(user: User) -> User:
    return single_item_task(user=user)


@task
def multiple_items_task(users: typing.List[User]) -> typing.List[User]:
    return users


@workflow
def multiple_items(users: typing.List[User]) -> typing.List[User]:
    return multiple_items_task(users=users)
```

Run:
```
pyflyte run bad_list_dataclass.py multiple_items --users '[{"github_handle":"enghabu"}]' 
```

Output:
<img width="1276" alt="image" src="https://github.com/user-attachments/assets/0219c043-8724-4116-8f5d-8bf73cf9235c">

### Expected behavior

However, running the single_item one:
```
pyflyte run bad_list_dataclass.py single_item --user '{"github_handle":"enghabu"}'  
```

Works normally.

### Are you sure this issue hasn't been raised already?

- [X] Yes

### Have you read the Code of Conduct?

- [X] Yes

**Comment by kumare3** [comment]
@EngHabu what version are you using, this should be fixed in 1.13

**Comment by Future-Outlier** [comment]
It's done, thank you @mao3267 !
